src/disco/bank/fd_txncache.c

```diff
@@ -324,61 +332,17 @@ fd_txncache_join( void * shtc ) {
     return NULL;
   }
 
-  return tc;
-}
-
-fd_txncache_t *
-fd_txncache_join_wksp( void * shtc ) {
-  if( FD_UNLIKELY( !shtc ) ) {
-    FD_LOG_WARNING(( "NULL shtc" ));
-    return NULL;
-  }
-
-  if( FD_UNLIKELY( !fd_ulong_is_aligned( (ulong)shtc, fd_txncache_align() ) ) ) {
-    FD_LOG_WARNING(( "misaligned shtc" ));
-    return NULL;
-  }
-
-  fd_txncache_t * tc = (fd_txncache_t *)shtc;
-
-  if( FD_UNLIKELY( tc->magic!=FD_TXNCACHE_MAGIC ) ) {
-    FD_LOG_WARNING(( "bad magic" ));
-    return NULL;
-  }
+  uchar * base = (uchar *)tc;
+  tc->root_slots = (ulong *)( base + tc->root_slots_off );
```

Comment:
?? This isn't right, the different joins share this memory still. You need to remove these pointers from the struct and just reference everything from the offset always (probably with helper fns, see eg, the pattern used by fd_cnc_app_laddr

https://github.com/firedancer-io/firedancer/blob/main/src/tango/cnc/fd_cnc.h#L211

Reply:
this change is made, testing against mainnet snapshot